src/randomizer.py

```diff
@@ -2080,6 +2080,102 @@ def visit_Decl(self, decl: ca.Decl) -> None:
     ast_util.insert_decl(fn, var, type, random)
 
 
+def perm_inline_get_structmember(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Creates an inline for accessing a struct member."""
+
+    typemap = build_typemap(ast)
+
+    ext_fns: List[str] = []
+    cands: List[ca.StructRef] = []
+
+    for item in ast.ext:
+        if (
+            isinstance(item, ca.Decl)
+            and isinstance(item.type, ca.FuncDecl)
+            and item.name
+        ):
+            ext_fns.append(item.name)
+        if isinstance(item, ca.FuncDef) and item.decl.name:
+            ext_fns.append(item.decl.name)
+
+    class Visitor(ca.NodeVisitor):
+        def visit_StructRef(self, node: ca.StructRef) -> None:
+            if region.contains_node(node):
+                cands.append(node)
+            self.generic_visit(node)
+
+    Visitor().visit(fn.body)
+    ensure(cands)
+
+    cand = random.choice(cands)
+    arg_type: SimpleType = decayed_expr_type(cand.name, typemap)
+    member_type: SimpleType = decayed_expr_type(cand, typemap)
+    member_name = cand.field.name
+
+    new_inline_fn_name = f"get_{member_name}"
+    counter = 1
+    while new_inline_fn_name in ext_fns:
+        counter += 1
+        new_inline_fn_name = f"get_{member_name}{counter}"
+
+    # Replace the StructRef with a FuncCall to the new inline yet to be created
+    replace_node(
+        fn.body, cand, ca.FuncCall(ca.ID(new_inline_fn_name), ca.ExprList([cand.name]))
+    )
+
+    assert isinstance(arg_type.type, ca.TypeDecl)
+    assert arg_type.type.declname
+
+    if isinstance(member_type, ca.PtrDecl):
+        assert isinstance(member_type.type, ca.TypeDecl)
+        assert isinstance(member_type.type.type, ca.IdentifierType)
+        member_type_str = member_type.type.type.names[0] + "*"
+    elif isinstance(member_type, ca.TypeDecl):
```

Comment:
passing a whole struct to an inline function feels unlikely... should we pass &x instead, or bail out?

Reply:
This isn't setup to pass a whole struct to an inline function, this is for the return value, which likely would not be a struct.

Reply:
```
inline s32 get_x8_spawnNum(Fighter *fighter2)
{
  return fighter2->x8_spawnNum;
}
```

example member is the spawnNum, in this case an s32